### PR TITLE
Apply Detector Scan calibration

### LINKED ISSUE #21327: ILL Powder Diffraction Detector Scan reduction
Similar to https://github.com/mantidproject/mantid/issues/19880, a new algorithm is required to treat detector scan reductions for the ILL powder diffractometers D2B and D20.

### PR DESCRIPTION
**Pull request by ianbush**
This algorithm applies a calibration file to a detector scan workspace. Please check the documentation for a full description.

**To test:**

Test with the attached files for D20. The monitors should remain unchanged, other elements in the input workspace should be scaled appropriately. There are 517 scan points, so the first 517 (monitors) are unchanged, the next 517 scaled by the first element in the calibration workspace, the next 517 by the second etc.

[D20_apply_calib_test.zip](https://github.com/mantidproject/mantid/files/1638463/D20_apply_calib_test.zip)

Fixes #21327.

**Release Notes** 

Release notes added.

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)?
- [ ] Are the unit tests small and test the class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] Do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Algorithms/src/SumOverlappingTubes.cpp [hunk 1]
```diff
@@ -237,17 +221,22 @@ void SumOverlappingTubes::getHeightAxis() {
     const auto &compAss = dynamic_cast<const ICompAssembly &>(*comp);
```
Comment (OwenArnold) [inline Framework/Algorithms/src/SumOverlappingTubes.cpp]: Is there a good reason you are not using `ComponentInfo` for this? @LamarMoore is reviewing some changes at the moment that give component info a `indexOfAny(const std::string& name` (but it's pretty much there now anyway`. and would be much cleaner than the implementation here.

Comment (OwenArnold) [reply Framework/Algorithms/src/SumOverlappingTubes.cpp]: No dynamic_cast needed for one thing.

**Review by OwenArnold — CHANGES_REQUESTED** [review]
Prelim review. I'd like to re-review after first round of changes made.

**Comment by ianbush** [comment]
@OwenArnold I maybe should have left this PR marked as in progress, but the `MergeRuns` changes are from PR #21456 and the `SumOverlappingTubes` from PR #21327 . I forgot they would show up here when I opened the PR. I'll look at making those changes on the other PRs.

**Comment by gvardany** [comment]
@OwenArnold I think all the dependencies for this are now merged, can you proceed please with the final review?

**Review by OwenArnold — APPROVED** [review]

**Pull request merged**